tutorials/advanced/transforms_tutorial.py

```diff
@@ -303,12 +316,98 @@
 #
 #     trans = ax.get_xaxis_transform()
 #
+# .. _transforms-fig-scale-dpi:
+#
+# Plotting in physical units
+# ==========================
+#
+# Sometimes we want an object to be a certain physical size on the plot.
+# Here we draw the same circle as above, but in physical units.  If done
+# interactively, you can see that changing the size of the figure does
+# not change the offset of the circle from the lower-left corner,
+# does not change its size, and the circle remains a circle regardless of
+# the aspect ratio of the axes.
+
+fig, ax = plt.subplots(figsize=(5, 4))
+x, y = 10*np.random.rand(2, 1000)
+ax.plot(x, y*10., 'go', alpha=0.2)  # plot some data in data coordinates
+# add a circle in fixed-units
+circ = mpatches.Circle((2.5, 2), 1.0, transform=fig.dpi_scale_trans,
+                      facecolor='blue', alpha=0.75)
+ax.add_patch(circ)
+plt.show()
+
+###############################################################################
+# If we change the figure size...
+
+fig, ax = plt.subplots(figsize=(7, 2))
+x, y = 10*np.random.rand(2, 1000)
+ax.plot(x, y*10., 'go', alpha=0.2)  # plot some data in data coordinates
+# add a circle in fixed-units
+circ = mpatches.Circle((2.5, 2), 1.0, transform=fig.dpi_scale_trans,
+                      facecolor='blue', alpha=0.75)
+ax.add_patch(circ)
+plt.show()
+
+###############################################################################
+# Another use is putting a patch with a set physical dimmension around a
+# data point on the axes.  Here we add together two transforms.  The
+# first sets the scaling of how large the ellipse should be and the second
+# sets its position.  The ellipse is then placed at the origin, and then
+# we use the helper transform :class:`~matplotlib.transforms.ScaledTranslation`
+# to move it
+# to the right place in the ``ax.transData`` coordinate system.
+# This helper is instantiated with::
+#
+#   trans = ScaledTranslation(xt, yt, scale_trans)
+#
+# where `xt` and `yt` are the translation offsets, and `scale_trans` is
+# a transformation which scales `xt` and `yt` at transformation time
+# before applying the offsets.
+#
+# Note the use of the plus operator on the transforms below.
+# This code says: first apply the scale transformation ``fig.dpi_scale_trans``
+# to make the Ellipse the proper size, but still centered at (0, 0),
+# and then translate the data to `xdata[0]` and `ydata[0]` in data space.
+#
+# In interactive use, the ellipse stays the same size even if the
+# axes limits are changed via zoom, but the ellipse's size doesn't change.
+#
+
+fig, ax = plt.subplots()
+xdata, ydata = (0.2, 0.7), (0.5, 0.5)
+ax.plot(xdata, ydata, "o")
+ax.set_xlim((0, 1))
+
+trans =  fig.dpi_scale_trans + \
+        transforms.ScaledTranslation(xdata[0], ydata[0], ax.transData)
+
+# plot an ellipse around the point that is 150 x 130 points in diameter...
+circle = mpatches.Ellipse((0, 0), 150/72, 130/72, angle=40, fill=None,
+        transform=trans)
```

Comment:
strange indent/spacing

Reply:
These typically get moved up to align with the opening `(` that it is in.